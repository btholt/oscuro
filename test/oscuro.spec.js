const oscuro = require("../");

test("reads one file", async () => {
  const ans = await oscuro(["./test/test1.json"]);
  expect(ans).toMatchSnapshot();
});

test("reads two files, both exist, returns first extant one", async () => {
  const ans = await oscuro(["./test/test1.json", "./test/test2.json"]);
  expect(ans).toMatchSnapshot();
});

test("reads two files, first exists, returns first extant one", async () => {
  const ans = await oscuro(["./test/test1.json", "./test/doesntExist.json"]);
  expect(ans).toMatchSnapshot();
});

test("reads two files, none exist, returns default", async () => {
  const ans = await oscuro(
    ["./test/doesntExist.json", "./test/alsoDoesntExist.json"],
    {
      defaultResponse: { success: true }
    }
  );
  expect(ans).toMatchSnapshot();
});

test("reads two files, second exists, returns first extant one", async () => {
  const ans = await oscuro(["./test/doesntExist.json", "./test/test1.json"]);
  expect(ans).toMatchSnapshot();
});

test("reads yaml", async () => {
  const ans = await oscuro(["./test/yamltest.yaml", "./test/test1.json"]);
  expect(ans).toMatchSnapshot();
});

test("reads js", async () => {
  const ans = await oscuro(["./test/jstest.js", "./test/test1.json"]);
  expect(ans).toMatchSnapshot();
});

test("handles js files not found", async () => {
  const ans = await oscuro(["./test/not-real.js", "./test/test1.json"]);
  expect(ans).toMatchSnapshot();
});

test("handles js paths", async () => {
  const ans = await oscuro([
    { path: "./test/jstest.js", key: "other" },
    "./test/test1.json"
  ]);
  expect(ans).toMatchSnapshot();
});

test("returns undefined when no default is given and no paths are found", async () => {
  const ans = await oscuro(["./not-real.json", "./also-not-real.json"]);
  expect(ans).toBeUndefined();
});

test("reads html", async () => {
  const ans = await oscuro(["./test/htmltest.html", "./also-not-real.json"]);
  expect(ans).toMatchSnapshot();
});

test("handles objects", async () => {
  const ans = await oscuro([
    { path: "./test/test1.json" },
    "./also-not-real.json"
  ]);
  expect(ans).toMatchSnapshot();
});

test("will read json as text", async () => {
  const ans = await oscuro([
    { path: "./test/test1.json", type: "text" },
    "./also-not-real.json"
  ]);
  expect(ans).toMatchSnapshot();
});

test("will read a dot file as json", async () => {
  const ans = await oscuro([{ path: "./test/.testdotfile", type: "json" }]);
  expect(ans).toMatchSnapshot();
});

test("will give you back only a key within the file", async () => {
  const ans = await oscuro([
    { path: "./test/test1.json", type: "json", key: "thing.cool.neat" }
  ]);
  expect(ans).toMatchSnapshot();
});

test("reads one file, returns path", async () => {
  const ans = await oscuro(["./test/test1.json"], { returnPathOnly: true });
  expect(ans).toBe("./test/test1.json");
});

test("reads two files, both exist, returns first extant one, returns path", async () => {
  const ans = await oscuro(["./test/test1.json", "./test/test2.json"], {
    returnPathOnly: true
  });
  expect(ans).toBe("./test/test1.json");
});

test("reads two files, second exists, returns first extant one, returns path", async () => {
  const ans = await oscuro(["./test/doesntExist.json", "./test/test1.json"], {
    returnPathOnly: true
  });
  expect(ans).toBe("./test/test1.json");
});

test("handles paths and keys when key exists", async () => {
  const pathObj = { path: "./test/test1.json", key: "thing.cool" };
  const ans = await oscuro([pathObj], {
    returnPathOnly: true
  });
  expect(ans).toBe(pathObj);
});

test("handles paths and keys when path exists but key doesn't", async () => {
  const pathObj = { path: "./test/test1.json", key: "not.real" };
  const ans = await oscuro([pathObj], {
    returnPathOnly: true
  });
  expect(ans).toBeUndefined();
});
