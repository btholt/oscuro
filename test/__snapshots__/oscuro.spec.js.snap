// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles js files not found 1`] = `
Object {
  "test": 2,
  "thing": Object {
    "cool": Object {
      "neat": Object {
        "bye": false,
        "hi": true,
      },
    },
  },
}
`;

exports[`handles js paths 1`] = `"cool stuff"`;

exports[`handles objects 1`] = `
Object {
  "test": 2,
  "thing": Object {
    "cool": Object {
      "neat": Object {
        "bye": false,
        "hi": true,
      },
    },
  },
}
`;

exports[`reads html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

<head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>Test For Oscuro </title>
</head>

<body>
    <h1>hi</h1>
</body>

</html>"
`;

exports[`reads js 1`] = `
Object {
  "other": "cool stuff",
}
`;

exports[`reads one file 1`] = `
Object {
  "test": 2,
  "thing": Object {
    "cool": Object {
      "neat": Object {
        "bye": false,
        "hi": true,
      },
    },
  },
}
`;

exports[`reads two files, both exist, returns first extant one 1`] = `
Object {
  "test": 2,
  "thing": Object {
    "cool": Object {
      "neat": Object {
        "bye": false,
        "hi": true,
      },
    },
  },
}
`;

exports[`reads two files, first exists, returns first extant one 1`] = `
Object {
  "test": 2,
  "thing": Object {
    "cool": Object {
      "neat": Object {
        "bye": false,
        "hi": true,
      },
    },
  },
}
`;

exports[`reads two files, none exist, returns default 1`] = `
Object {
  "success": true,
}
`;

exports[`reads two files, second exists, returns first extant one 1`] = `
Object {
  "test": 2,
  "thing": Object {
    "cool": Object {
      "neat": Object {
        "bye": false,
        "hi": true,
      },
    },
  },
}
`;

exports[`reads yaml 1`] = `
Object {
  "cool": Object {
    "thing": 5,
  },
  "test": true,
}
`;

exports[`will give you back only a key within the file 1`] = `
Object {
  "bye": false,
  "hi": true,
}
`;

exports[`will read a dot file as json 1`] = `
Object {
  "success": true,
  "testing": "dot file",
}
`;

exports[`will read json as text 1`] = `
"{
  \\"test\\": 2,
  \\"thing\\": {
    \\"cool\\": {
      \\"neat\\": {
        \\"hi\\": true,
        \\"bye\\": false
      }
    }
  }
}
"
`;
